llvm/lib/Target/AMDGPU/AMDGPU.td

```diff
@@ -167,6 +167,10 @@ def FeatureCuMode : SubtargetFeature<"cumode",
   "Enable CU wavefront execution mode"
 >;
 
+def FeaturePreciseMemory
```

Comment:
I'm not a fan of using a feature for this, I think we should have a backend CL option instead. After all this isn't an architecture feature but just a behavior change for SIMemoryLegalizer.

Reply:
End user needs this, so a clang command-line option has been added. There was a lot discussion on this. I'll forward you the email.

Reply:
Understood :) 
Can you remove the `amdgpu` prefix from the option? All target features are already specific to AMDGPU, e.g. xnack, sramecc, etc. so you can just use something like `precise-memory`

Reply:
The name was the result of some discussions last year. I've forwarded you the email.

Reply:
The subtarget feature prefix should be removed. The subtarget feature name is not the user facing component of this, and no user should be directly using it

Reply:
As it is, we have a clang command-line option "-mamdgpu-precise-memory-op". When specified, "+amdgpu-precise-memory-op" would be passed to the backend as a target feature. So if the backend uses a different name for this feature, e.g., dropping the 'amdgpu' prefix as you suggested, the target feature passed by clang will be considered invalid and silently ignored.

In short, my understanding is that if the prefix is dropped in the backend (in `AMDGPU.td`), then the clang command-line option also has to drop it. If you know how to keep it in clang but drop it in the backend, pls let me know.

Reply:
Thanks for the suggestion. It looks it would work. However, I'm not sure the extra overhead is worth it just for having a shorter name in the backend. After all this is not going to be a commonly used option, and the name in the backend is not exposed to end user.

Reply:
The extra overhead is just 3 lines in `clang/lib/Driver/ToolChains/AMDGPU.cpp`, it's negligible.
We don't have any target feature that starts with `amdgpu` so it's out of place IMO, it should really drop the `amdgpu-` prefix.

Reply:
The function `hasFlag()` actually appears to be pretty expensive, and it's going to be called every time, regardless of whether the flag is used or not.

Let's get Matt's opinion on this. @arsenm What do you think?

Reply:
It's only called once per run by the driver, yes

We already do this for wavefrontsize64, and pretty much all other options that aren't automatically handled. We even call `hasFlag` twice on the same option in some cases instead of saving the result somewhere, so I doubt `hasFlag` can be considered expensive enough to justify not dropping `amdgpu-` from the feature name

Reply:
@Pierre-vh With the suggested change, the func `getAMDGPUTargetFeatures` looks something like the following:
```
void amdgpu::getAMDGPUTargetFeatures(...) {
...
  if (Args.hasFlag(options::OPT_mwavefrontsize64,
                   options::OPT_mno_wavefrontsize64, false))
    Features.push_back("+wavefrontsize64");

  if (Args.hasFlag(options::OPT_mamdgpu_precise_memory_op,
                   options::OPT_mno_amdgpu_precise_memory_op, false)) {
    Features.push_back("+precise-memory");
  }
  handleTargetFeaturesGroup(D, Triple, Args, Features,
                            options::OPT_m_amdgpu_Features_Group);
}

However, `handleTargetFeaturesGroup` does not seem to care whether an Arg is claimed or not. It will process every Arg, and we end up with the following:
`"-target-feature" "+precise-memory" "-target-feature" "+amdgpu-precise-memory-op"`

Reply:
A possible fix is to check `hasFlag` after calling `handleTargetFeaturesGroup`, and if the check is true, replace the string `"+amdgpu-precise-memory-op"` in the `Features` vector with `"+precise-memory"`, as follows:
```
  handleTargetFeaturesGroup(D, Triple, Args, Features,
                            options::OPT_m_amdgpu_Features_Group);

  if (Args.hasFlag(options::OPT_mamdgpu_precise_memory_op,
                   options::OPT_mno_amdgpu_precise_memory_op, false)) {
    for (auto It = Features.begin(); It != Features.end(); ++It) {
      if (*It == "+amdgpu-precise-memory-op")
        *It = StringRef("+precise-memory");
    }
```
@Pierre-vh Your thoughts?

Reply:
Just remove `m_amdgpu_Features_Group` from your option's `SimpleMFlag`, follow the same pattern as `wavefrontsize64`

Reply:
Thanks! This works. Code has been updated. Pls take a look.